hephaistus/main-theme/distribution-system/chord/playground/nullist0/lib/src/main/kotlin/org/lia/dht/nullist0/chord/model/ChordScope.kt

```diff
@@ -0,0 +1,37 @@
+package org.lia.dht.nullist0.chord.model
+
+/**
+ * An interface to build identifier of Chord Ring.
+ */
+abstract class ChordScope<Id, Value> {
+    abstract fun AbstractChordNode<Id, Value>.toId(): Id
+
+    abstract fun bitSize(): Int
+
+    abstract fun Int.toId(): Id
+
+    /**
+     * The [Id] in a chord ring, is an element of integers modulo n, where a natural number n is a power of 2.
+     * There is a natural addition in such set, so this function will implement that addition.
+     */
+    abstract infix operator fun Id.plus(other: Id): Id
+
+    /**
+     * The [Id] in a chord ring, is an element of integers modulo n, where a natural number n is a power of 2.
+     * Since there is no order in the such set, this function will implement an order on the integer set,
+     * not on the modulo set. In other words, this will return the natural order of integers.
+     */
+    protected abstract infix operator fun Id.compareTo(other: Id): Int
+
+    infix fun Id.inOpenInterval(pair: Pair<Id, Id>): Boolean  {
+        val id = this@inOpenInterval
+        val (start, end) = if (pair.first < pair.second) pair else pair.second to pair.first
+        return start < id && id < end
```

Comment:
IMHO, in 2^3 size of Id, id is 2, first is 7 and second is 4, can not return true. (or also fail with id is 8)

Reply:
Thanks for comment. I missed. I will fix this.

Reply:
I fixed at 480d1ec45676ffffb4a15274350bf79491b4cd15, and add some test cases.